core/editor/data/favorites/manager.php

```diff
@@ -0,0 +1,278 @@
+<?php
+namespace Elementor\Core\Editor\Data\Favorites;
+
+use Elementor\Core\Editor\Data\Favorites\Collections\Widgets;
+use WP_Error;
+
+class Manager {
+
+	/**
+	 * List of registered collections.
+	 *
+	 * @var Collection[]
+	 */
+	protected $collections = [];
+
+	/**
+	 * The name of the merge action.
+	 *
+	 * @var string
+	 */
+	const ACTION_MERGE = 'merge';
+
+	/**
+	 * The name of the delete action.
+	 *
+	 * @var string
+	 */
+	const ACTION_DELETE = 'delete';
+
+	/**
+	 * Manager constructor.
+	 */
+	public function __construct() {
+		// Register default collections
+		$this->register( Widgets::class );
+		// ...
+
+		$this->populate();
+	}
+
+	/**
+	 * Get user favorites collection.
+	 *
+	 * @param string[]|string $collections
+	 *
+	 * @return array
+	 */
+	public function get( $collections = null ) {
+		if ( null === $collections ) {
+			return $this->compile();
+		}
+
+		return $this->implementation( $collections )
+			->values();
+	}
+
+	/**
+	 * Merge new user favorites to collection.
+	 *
+	 * @param string $collection
+	 * @param array  $favorites
+	 * @param bool   $store
+	 *
+	 * @return array|bool
+	 */
+	public function merge( $collection, $favorites, $store = true ) {
+		return $this->update( $collection, $favorites, static::ACTION_MERGE, $store );
+	}
+
+	/**
+	 * Delete existing favorites from collection.
+	 *
+	 * @param string $collection
+	 * @param array  $favorites
+	 * @param bool   $store
+	 *
+	 * @return array|int
+	 */
+	public function delete( $collection, $favorites, $store = true ) {
+		return $this->update( $collection, $favorites, static::ACTION_DELETE, $store );
+	}
+
+	/**
+	 * Update favorites collection by merging or deleting from it.
+	 *
+	 * @param      $collection
+	 * @param      $favorites
+	 * @param      $action
+	 * @param bool $store
+	 *
+	 * @return array|boolean
+	 */
+	public function update( $collection, $favorites, $action, $store = true ) {
+		$implementation = $this->implementation( $collection );
+		$favorites = $implementation->prepare( $favorites );
+
+		switch ( $action ) {
+			case static::ACTION_MERGE:
+				$implementation->merge( $favorites );
+				break;
+			case static::ACTION_DELETE:
+				$implementation->filter(
+					function( $value ) use ( $favorites ) {
+						return ! in_array( $value, $favorites, true );
+					}
+				);
+				break;
+			default:
+				$this->action_doesnt_exists( $action );
+		}
+
+		if ( $store && ! $this->store() ) {
+			return false;
+		}
+
+		return $implementation->values();
+	}
+
+	/**
+	 * Check whether favorites collection is registered.
+	 *
+	 * @param string $collection
+	 *
+	 * @return bool
+	 */
+	public function exists( $collection ) {
```

Comment:
```suggestion
	public function exists( $collection_key ) {
```
OR
```suggestion
	public function exists( $key ) {
```

Reply:
That's confusing, because I used `$collection` as the name of every argument that represents collection key in the class, and this method's argument shouldn't be the ugly duckling. The reason is that a good argument name is one who is short and sweet, because in long method declarations (like `action`) it saves space, and makes the code more readable. If there was an instance in the class where `$collection` is a representation of other than the collection key, the situation was different. Tell me please if you disagree.